endpoints/openrtb2/amp_auction.go

```diff
@@ -307,7 +307,7 @@ func (deps *endpointDeps) parseAmpRequest(httpRequest *http.Request) (req *openr
 
 	// At this point, we should have a valid request that definitely has Targeting and Cache turned on
 
-	e = deps.validateRequest(req)
+	e = deps.validateRequest(&openrtb_ext.RequestWrapper{Request: req})
```

Comment:
I know this is not the goal of this PR but what do you think about ultimately wrapping the request in middleware so the endpoint handlers only ever work with the wrapper? I think that will reduce code complexity and cognitive load a bit.

Reply:
We may still need the concept of the simple request, as that can have only one state for the ext fields. With the wrapper, once you have dirty bits, the state of the original request is no longer in sync with the ext objects, so the state is a bit more undefined until a sync is called to push all the changes through. I am not 100% on this though.